### PR TITLE
feat(helm): values override as arguments

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-931

Capture d’écran 2023-12-27 à 11 56 58

+ Update `<EmptyState />` component with current design

Capture d’écran 2023-12-27 à 11 50 35

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Don’t Sacrifice Readability, you are going to debug it in 6 months

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(helm): values override as arguments - 2023-12-22T15:59:37Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/d7c59eae-6407-4f15-88e2-d9345b226720/services/general)

 Another comment will be posted when deployments are terminated

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://1061-console-staging.zc531a994.rustrocks.cloud)

File: libs/shared/ui/src/lib/components/inputs/input-select-small/input-select-small.tsx [hunk 1]
```diff
@@ -52,7 +54,10 @@ export function InputSelectSmall(props: InputSelectSmallProps) {
         data-testid={dataTestId || 'input-select-small'}
         name={name}
         value={value}
-        className={`input input__select--small ${inputClassName}`}
+        className={twMerge(
+          'h-9 px-4 py-2 cursor-pointer pl-2 pr-6 pt-2 pb-2 bg-neutral-100 border border-neutral-250 text-neutral-400 rounded text-sm appearance-none w-full',
+          inputClassName
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/inputs/input-select-small/input-select-small.tsx:59]: Remove `input input__select--small` to use tailwind classes only

File: libs/domains/service-helm/feature/src/lib/values-override-files-setting/__snapshots__/values-override-files-setting.spec.tsx.snap [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`ValuesOverrideYamlSetting should match snapshot with GIT_REPOSITORY type 1`] = `
```
Comment (RemiBonnet) [inline libs/domains/service-helm/feature/src/lib/values-override-files-setting/__snapshots__/values-override-files-setting.spec.tsx.snap:3]: naming error

File: libs/pages/services/src/lib/feature/page-helm-create-feature/step-summary-feature/step-summary-feature.tsx [hunk 3]
```diff
@@ -265,9 +283,13 @@ export function StepSummaryFeature() {
...
           <div className="flex-grow mr-2">
-            <div className="text-sm text-neutral-400 font-bold mb-5">
-              {networkingData.ports.length} {pluralize(networkingData.ports.length, 'services')} exposed publicly
-            </div>
+            {networkingData.ports.length > 0 ? (
+              <div className="text-sm text-neutral-400 font-bold mb-5">
+                {networkingData.ports.length} {pluralize(networkingData.ports.length, 'service')} exposed publicly
+              </div>
+            ) : (
+              <span className="text-sm text-neutral-400 font-bold">No service exposed</span>
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/feature/page-helm-create-feature/step-summary-feature/step-summary-feature.tsx:291]: Add this one to be iso with Figma when we don't have Network

File: libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+export interface HelmValuesArgumentsData {
+  arguments: {
+    key: string
+    type: ArgumentTypes
+    value: string
+    json?: string
+  }[]
+}
+
+export interface ValuesOverrideArgumentsSettingProps extends PropsWithChildren {
```
Comment (ctjhoa) [inline libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx:33]:
```suggestion
export interface ValuesOverrideArgumentsSettingProps extends PropsWithoutChildren {
```

File: libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+function Row({ index, remove }: { index: number; remove: UseFieldArrayRemove }) {
+  const { watch, control } = useFormContext()
+
+  const [openEditor, setOpenEditor] = useState(true)
+  const valueTypeJson = watch(`arguments.${index}.type`) === 'json'
+
+  return (
+    <li className="mb-3 last:mb-0">
```
Comment (ctjhoa) [inline libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx:45]: Remaining props should be spread on `li` to pass `key` and other props

File: libs/pages/application/src/lib/feature/page-settings-values-override-arguments-feature/page-settings-values-override-arguments-feature.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import {
+  type ArgumentTypes,
+  type HelmValuesArgumentsData,
+  ValuesOverrideArgumentsSetting,
+} from '@qovery/domains/service-helm/feature'
+import { useEditService, useService } from '@qovery/domains/services/feature'
+import { Button } from '@qovery/shared/ui'
+import { buildEditServicePayload } from '@qovery/shared/util-services'
+
+export const generateArguments = (
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-values-override-arguments-feature/page-settings-values-override-arguments-feature.tsx:12]:
```suggestion
const generateArguments = (
```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-values-override-arguments-feature/page-settings-values-override-arguments-feature.tsx (hunk 6 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-values-override-arguments-feature/page-settings-values-override-arguments-feature.tsx]: https://github.com/Qovery/console/pull/1061/commits/038e701b38c313396c3bbe5ef0f7da2fc410f311

File: libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx (hunk 5 above)
Comment (RemiBonnet) [reply libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx]: https://github.com/Qovery/console/pull/1061/commits/038e701b38c313396c3bbe5ef0f7da2fc410f311

File: libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx (hunk 4 above)
Comment (RemiBonnet) [reply libs/domains/service-helm/feature/src/lib/values-override-arguments-setting/values-override-arguments-setting.tsx:33]: It's correct with `PropsWithChildren`? `PropsWithoutChildren` doesn't exist

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**